servers/jax-rs-testextension/src/main/java/org/projectnessie/jaxrs/ext/NessieJaxRsExtension.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.jaxrs.ext;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static org.projectnessie.services.config.ServerConfigExtension.SERVER_CONFIG;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.DirectoryNotEmptyException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.function.Supplier;
+import javax.ws.rs.core.Application;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.jboss.weld.environment.se.Weld;
+import org.jboss.weld.environment.se.WeldContainer;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.projectnessie.jaxrs.NessieJaxRsJsonMappingExceptionMapper;
+import org.projectnessie.jaxrs.NessieJaxRsJsonParseExceptionMapper;
+import org.projectnessie.services.authz.AccessCheckerExtension;
+import org.projectnessie.services.config.ServerConfigExtension;
+import org.projectnessie.services.rest.ConfigResource;
+import org.projectnessie.services.rest.ContentsKeyParamConverterProvider;
+import org.projectnessie.services.rest.ContentsResource;
+import org.projectnessie.services.rest.InstantParamConverterProvider;
+import org.projectnessie.services.rest.NessieExceptionMapper;
+import org.projectnessie.services.rest.TreeResource;
+import org.projectnessie.services.rest.ValidationExceptionMapper;
+import org.projectnessie.versioned.PersistVersionStoreExtension;
+import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
+
+/** A JUnit 5 extension that starts up Weld/JerseyTest. */
+public class NessieJaxRsExtension implements BeforeAllCallback, AfterAllCallback {
```

Comment:
This will replace the "old `NessieJaxRsExtension`" and with the changed package name help to prevent potential packaging issues with JPMS (Quarkus already warns about duplicate packages from different modules).

Reply:
If we have to interfere with JAX-RS setup and dependency injection for these tests, why not rely on Quarkus to manage that and restructure the affected tests to use an external (forked) Quarkus server?

Reply:
There's intentionally no Quarkus in all these JAX-RS modules.

Reply:
In production code - yes, but why can we not use Quarkus for testing JAX-RS stuff? It is part of the Quarkus runtime, is it not? We seem to have quarkus-specific test classes that validate the same code anyway... or am I confused?

Reply:
Indeed, I got confused. I see now that this is a completely separate runtime env.